web/app/suoritus/KoulutusmoduuliEditor.jsx

```diff
@@ -33,13 +34,25 @@ export const KoulutusmoduuliEditor = ({model}) => {
 }
 
 const TunnisteEditor = ({model}) => {
-  const overrideEdit = model.context.editAll ? true : false
+  const overrideEdit = (model.context.editAll || (model.context.edit && isMuutaAmmatillistaPäätasonSuoritus(model.context.suoritus))) ? true : false
   const päätasonsuoritus = model.context.suoritus
   const tyyppi = suorituksenTyyppi(päätasonsuoritus)
   const käytäPäätasonSuoritusta =
     ['aikuistenperusopetuksenoppimaara', 'aikuistenperusopetuksenoppimaaranalkuvaihe'].includes(tyyppi) || model.value.classes.includes('lukionoppiaineidenoppimaarat2019')
+  const tutkinnonNimiModel = tutkinnonNimi(model)
+  const excludedProperties = ["koodistoUri"]
+
+  tutkinnonNimiModel.value.properties && tutkinnonNimiModel.value.properties.forEach(p => {
+    p.editable = p.editable && (!excludedProperties.includes(p.key) || !isMuutaAmmatillistaPäätasonSuoritus(model.context.suoritus))
+  })
```

Comment:
Tämä tutkinnonNimiModel kannattaisi alustaa jo ylemmässä komponentissa oikeaan muotoon, jos vanhempi tarvitsee tuon editable-propertyn omaan käyttöönsä. Jos se ei tarvitse, niin tämä kannattaisi tehdä mieluummin näin, ettei vahingossa mutatoida propertya:

```
const editableTutkinnonNimiModel = tutkinnonNimiModel.value.properties && tutkinnonNimiModel.value.properties.map(p => {
    ...p,
    editable: p.editable && (!excludedProperties.includes(p.key) || !isMuutaAmmatillistaPäätasonSuoritus(model.context.suoritus))
  })
```